apiserver/facades/client/modelconfig/modelconfig.go

```diff
@@ -194,6 +219,33 @@ func (c *ModelConfigAPI) ModelSet(ctx context.Context, args params.ModelSet) err
 	return err
 }
 
+// setAgentStream is responsible for setting the agent stream to use on the
+// current model. This exists because the way ask users to control this value is
+// still via model config as a user interface. If the value of agent stream
+// passed to this function is an empty string no operation will be performed.
+func (s *ModelConfigAPI) setAgentStream(ctx context.Context, agentStream string) error {
+	if agentStream == "" {
+		return nil
+	}
+
+	s.logger.Debugf(ctx, "setting agent stream to %q via model config", agentStream)
+
+	err := s.modelAgentService.SetModelAgentStream(
+		ctx, agentbinary.AgentStream(agentStream),
+	)
+	if errors.Is(err, coreerrors.NotValid) {
+		return internalerrors.Errorf(
+			"agent stream %q is not a valid value", agentStream,
+		).Add(coreerrors.NotValid)
+	} else if err != nil {
+		return internalerrors.Errorf(
+			"setting agent stream to value %q for model: %w", agentStream, err,
```

Comment:
This will stutter a bit and sounds clumsy
`setting agent stream to value "proposed" for model: setting model agent stream "released" to value 2 in state: db offline`

Can we tweak the messages to improve readability

Reply:
Done & Done